src/6_branch.js

```diff
@@ -1958,10 +1958,15 @@ Branch.prototype['setAPIResponseCallback'] = wrap(callback_params.NO_CALLBACK, f
 Branch.prototype['setDMAParamsForEEA'] = wrap(callback_params.CALLBACK_ERR, function(done, eeaRegion, adPersonalizationConsent, adUserDataUsageConsent) {
 	try {
 		var dmaObj = {};
-		dmaObj.eeaRegion = eeaRegion || false;
-		dmaObj.adPersonalizationConsent = adPersonalizationConsent || false;
-		dmaObj.adUserDataUsageConsent = adUserDataUsageConsent || false;
-		this._storage.set('branch_dma_data', safejson.stringify(dmaObj), true);
+		if (eeaRegion !== null && eeaRegion !== undefined) {
```

Comment:
How is this method called in the first place?

Reply:
I am starting too see what is our problem, initially

Reply:
It does seem this is good place to inspect each parameters, and provide meaningful validation warning, and make necessarily noop, if required. Let's refine this method a bit. 

I would also like us to update 1_utils.js so that it doesn't make any assumptions and make it simpler to follow

Reply:
```js
/**
 * Check if given value is boolean or not 
 * 
 * @param {any | null | undefined} [value]
 * @returns {boolean}
 */
utils.isBoolean = function (value) {
	if (value === null) {
		return false;
	}

	if (value === undefined) {
		return false;
	}

	return value === true || value == false;
}

```

Reply:
```js
	try {
		var valid = true;
		if (utils.isBoolean(eeaRegion)) {
			console.warn("eeaRegion must be non-null boolean");
			valid = false;
		}

		if (utils.isBoolean(adPersonalization)) {
			console.warn("eeaRegion must be non-null boolean");
			valid = false;
		}

		if (utils.isBoolean(adUserData)) {
			console.warn("eeaRegion must be non-null boolean");
			valid = false;
		}

		if (!valid) {
			return
		}

		var dmaObj = {
			eea: eea,
			adPersonalization: adPersonalization,
			adUserData: adUserData,
		};

		this._storage.set('branch_dma_data', safejson.stringify(dmaObj), true);
	}
	catch (e) {
		console.error("setDMAParamsForEEA::An error occured while setting DMA parameters for EEA", e);
	}
	done();
```

Reply:
```js
	try {
		var valid = true;
		if (utils.isBoolean(eeaRegion)) {
			console.warn("eeaRegion must be non-null boolean");
			valid = false;
		}

		if (utils.isBoolean(adPersonalization)) {
			console.warn("eeaRegion must be non-null boolean");
			valid = false;
		}

		if (utils.isBoolean(adUserData)) {
			console.warn("eeaRegion must be non-null boolean");
			valid = false;
		}

		if (!valid) {
			return
		}

		var dmaObj = {
			eea: eea,
			adPersonalization: adPersonalization,
			adUserData: adUserData,
		};

		this._storage.set('branch_dma_data', safejson.stringify(dmaObj), true);
	}
	catch (e) {
		console.error("setDMAParamsForEEA::An error occured while setting DMA parameters for EEA", e);
	}
	done();
```

``js
/**
 * Checks if given endpoint is v1
 * @param {string | null | undefined} endpoint
 * @returns {boolean}
 */
utils.isV1Endpoint = function(endpoint) {
	if (!endpoint) {
		return false;
	}

	return endpoint.includes('/v1/open') || endpoint.includes('/v1/pageview');
};

/**
 * Checks if given endpoint is v2
 * @param {string | null | undefined} endpoint
 * @returns {boolean}
 */
utils.isV2Endpoint = function(endpoint) {
	if (!endpoint) {
		return false;
	}

	return endpoint.includes('/v2/event/standard/') || endpoint.includes('/v2/event/standard');
};
```

Reply:
Let's not complicate our methods like that. I am seeing that we are making two things
1. iterate through these params 
2. depending on each methods, we are actually entering separate code block, within iteration.

Reply:
This is classic case of method cluttering with over extended abstraction